components/Modal/index.tsx

```diff
@@ -0,0 +1,56 @@
+import DistrictData from '../../data/districts.json';
+
+type props = {
+  data: React.MouseEvent<SVGPathElement> | null;
+};
+
+type districtDetails = {
+  label: string;
+  value: string | number;
+}
+
+
+const Modal = (props: props) => {
+  const { data } = props || {}
+  const { pageY, pageX, target } = data || {}
+  const district = DistrictData[(target as any)?.id]
```

Comment:
If you're using just pageX, pageY and target from the event, why don't you just pass them as props instead of passing the whole event?

Reply:
Also take a look at how other components are declared.
Let's keep it consistent.